### PR TITLE
Add `pbxproject_targets` generator

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

**Comment by brentleyjones** [comment]
I need to add a couple more tests (marked with `FIXME:`), but this is ready to review now.

File: tools/generators/pbxproject_targets/README.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+## Output
+
+Here is an example output:
+
+### `pbxproject_targets`
+
+```
+
```
Comment (brentleyjones) [inline tools/generators/pbxproject_targets/README.md:117]: Me: add the examples here.

**Comment by brentleyjones** [comment]
Starlark half if it helps with review: https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2354

**Review by thiagohmcruz — APPROVED** [review]
It's a big change so haven't gone through every detail, payed attention to the structure, some code paths that caught my attention, tests and comments. Overall seems similar to other generators so let me know if you want feedback about something more specific in this particular generator, happy to 👀!

File: tools/generators/pbxproject_targets/src/Generator/IdentifyTargets.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+extension Generator.IdentifyTargets {
+    public typealias Callable = (
+        _ consolidationMapArguments: [ConsolidationMapArguments],
+        _ logger: Logger,
+        _ consolidateTargets: Generator.ConsolidateTargets,
+        _ disambiguateTargets: Generator.DisambiguateTargets,
+        _ innerIdentifyTargets: Generator.InnerIdentifyTargets
+    ) throws -> [IdentifiedTarget]
+
+    // TODO: Add test
```
Comment (brentleyjones) [inline tools/generators/pbxproject_targets/src/Generator/IdentifyTargets.swift:64]: Last test technically needed, but I'm too tired to do it, and it's low value imo.

**Review by luispadron — APPROVED** [review]

**Pull request merged**